crates/shared/src/price_estimation/trade_verifier.rs

```diff
@@ -400,16 +414,86 @@ fn encode_settlement(
         &query.in_amount.get(),
     );
 
-    EncodedSettlement {
+    let mut trades = vec![encoded_trade];
+
+    for jit_order in trade.jit_orders.iter() {
+        let order_data = OrderData {
+            sell_token: jit_order.sell_token,
+            buy_token: jit_order.buy_token,
+            receiver: Some(jit_order.receiver),
+            sell_amount: jit_order.sell_amount,
+            buy_amount: jit_order.buy_amount,
+            valid_to: jit_order.valid_to,
+            app_data: AppDataHash::from_str(&jit_order.app_data)?,
+            fee_amount: 0.into(),
+            kind: match &jit_order.side {
+                dto::Side::Buy => OrderKind::Buy,
+                dto::Side::Sell => OrderKind::Sell,
+            },
+            partially_fillable: false,
+            sell_token_balance: jit_order.sell_token_source,
+            buy_token_balance: jit_order.buy_token_destination,
+        };
+        let (owner, signature) = match jit_order.signing_scheme {
+            SigningScheme::Eip1271 => {
+                let (owner, signature) = jit_order.signature.split_at(20);
+                let owner = H160::from_slice(owner);
+                let signature = Signature::from_bytes(jit_order.signing_scheme, signature)?;
+                (owner, signature)
+            }
+            SigningScheme::PreSign => {
+                let owner = H160::from_slice(&jit_order.signature);
+                let signature =
+                    Signature::from_bytes(jit_order.signing_scheme, Vec::new().as_slice())?;
+                (owner, signature)
+            }
+            _ => {
+                let signature =
+                    Signature::from_bytes(jit_order.signing_scheme, &jit_order.signature)?;
+                let owner = signature
+                    .recover(domain_separator, &order_data.hash_struct())?
+                    .unwrap()
+                    .signer;
+                (owner, signature)
+            }
+        };
+
+        tokens.push(jit_order.sell_token);
+        tokens.push(jit_order.buy_token);
+
+        match jit_order.side {
+            dto::Side::Sell => {
+                clearing_prices.push(jit_order.buy_amount);
+                clearing_prices.push(jit_order.executed_amount);
```

Comment:
@harisang in this PR we are testing that quotes can be settled (and verified) against JIT orders. We only introduced the concept of JIT orders in the `/quote` response because solvers could also settled against user orders by indexing the orderbook and pretending the user order is a JIT order.
In this particular part this would cause JIT orders to be settled at the limit price in the settlement. So if a solver returns a regular user order as a JIT order which would not reflect what would happen in a real settlement entirely correctly. I assumed this should be fine but I actually never asked the solver team about it.
Do you foresee any significant issues from this detail? Asking because we'd otherwise have to be able to know which JIT orders are actually surplus capturing in the quote verification which would be a huge PITA.

Reply:
In the competition encoding, executed amount is not used here:
https://github.com/cowprotocol/services/blob/559ebf3dce53837f660de743bbb55523e4e3a1c9/crates/driver/src/domain/competition/solution/encoding.rs#L116-L125

Maybe that is why the e2e test requires settlement contract to be funded with DAI:
https://github.com/cowprotocol/services/blob/8fd8cf60c823a024f9d6718634ece600124b17b4/crates/e2e/tests/e2e/cow_amm.rs#L1208-L1210

But when using plain sell/buy amounts, the simulation gets reverted with a vague error, so I couldn't understand what exactly is wrong with that case.